aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -68,7 +68,67 @@
 #include <numeric>
 #include <vector>
 
-namespace at { namespace native {
+namespace at {
+namespace meta {
+
+native::SCATTER_GATHER_OP get_operator_enum(const std::string& reduce) {
+  if (reduce == "add") {
+    return native::SCATTER_GATHER_OP::REDUCE_ADD;
+  } else if (reduce == "multiply") {
+    return native::SCATTER_GATHER_OP::REDUCE_MULTIPLY;
+  } else {
+    TORCH_CHECK(false, "reduce argument must be either add or multiply.");
+  }
+}
+
+template <typename Meta>
+void scatter_meta_impl(
+    Meta& meta,
+    const Tensor& self,
+    int64_t dim,
+    const Tensor& index,
+    const c10::optional<std::string>& reduce) {
+  TORCH_CHECK_INDEX(
+      index.scalar_type() == ScalarType::Long,
+      "scatter_(): Expected dtype int64 for index.");
+
+  at::assert_no_internal_overlap(self);
+  at::assert_no_overlap(self, index);
+  meta.set_output(self.sizes(), self.options());
+
+  if (reduce.has_value()) {
+    TORCH_CHECK(
+        at::isFloatingType(self.scalar_type()) ||
```

Comment:
why is this outrightly disallowed for integers?

Reply:
I believe it's a problem on the CUDA side. Apparently, it depends on `gpuAtomicMul` implemented only for floating point types (not sure why, though) in [_THCAtomics.cuh_][1]. That said, I think I could, at least, enable it for CPUs.

[1]: https://github.com/pytorch/pytorch/blob/master/aten/src/THC/THCAtomics.cuh